server/mailet/integration-testing/src/main/java/org/apache/james/mailets/configuration/SmtpConfiguration.java

```diff
@@ -155,7 +174,17 @@ public String serializeAsXml() throws IOException {
         scopes.put("verifyIdentity", verifyIndentity.toString());
         scopes.put("maxmessagesize", maxMessageSize);
         scopes.put("bracketEnforcement", bracketEnforcement);
-        scopes.put("hooks", addittionalHooks);
+
+        List<Map<String, Object>> additionalHooksWithConfig = additionalHooks.stream()
+                .map(hook -> {
+                    Map<String, Object> hookScope = new HashMap<>();
+                    hookScope.put("hookFqcn", hook.hookFqcn);
+                    hookScope.put("hookConfigAsXML", hook.hookConfig.entrySet());
+                    return hookScope;
+                })
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                .collect(ImmutableList.toImmutableList());
```

(Immutable collections for the <3 )